config/formats.js

```diff
@@ -562,6 +562,36 @@ let Formats = [
 		ruleset: ['Pokemon', 'Standard NEXT', 'Team Preview'],
 		banlist: ['Uber'],
 	},
+	{
+		name: '[Gen 7] Shitmons',
+		threads: [
+			`&bullet; <a href="http://spo.ink/shitmons">Shitmons</a>`,
+		],
+
+		gameType: 'doubles',
+		teamLength: {
+			validate: [2, 2],
+			battle: 2,
+		},
+		ruleset: ['Cancel Mod', 'HP Percentage Mod'],
+		banlist: [
+			'Honedge', 'Doublade', 'Aegislash', 'Kitsunoh', 'Shedinja ++ Sturdy', 'Power Construct', 'Schooling', 'Iron Barbs', 'Rough Skin',
+			'Pure Power', 'Huge Power', 'Cursed Body', 'Moody', 'Pressure', 'Wonder Guard', 'Fur Coat', 'Parental Bond', 'Protean', 'Stamina',
+			'Snow Warning', 'Sand Stream', 'Primordial Sea', 'Desolate Land', 'Poison Heal', 'Ice Body', 'Rain Dish', 'Grassy Surge',
+			'Cheek Pouch', 'Aguav Berry', 'Berry', 'Berry Juice', 'Black Sludge', 'Enigma Berry', 'Figy Berry', 'Iapapa Berry', 'Leftovers',
+			'Mago Berry', 'Oran Berry', 'Sitrus Berry', 'Wiki Berry', 'Rocky Helmet', 'Normalium-Z', 'Aerodactylite', 'Aggronite', 'Ampharosite',
+			'Blastoisinite', 'Blazikenite', 'Charizardite X', 'Charizardite Y', 'Diancite', 'Galladite', 'Garchompite', 'Gardevoirite',
+			'Gyaradosite', 'Latiasite', 'Latiosite', 'Lucarionite', 'Metagrossite', 'Salamencite', 'Sceptilite', 'Steelixite', 'Swampertite',
+			'Tyranitarite', 'Venusaurite',
+		],
+		onValidateSet: function (set) {
+			let template = this.getTemplate(set.species);
+			let bst = template.baseStats.hp + template.baseStats.atk + template.baseStats.def + template.baseStats.spa + template.baseStats.spd + template.baseStats.spe;
+			if (set.name === template.baseSpecies) return [`${set.species} does not have a nickname set.`, `(Edgy nicknames are required in Shitmons)`];
+			if (bst > 600) return [`${template.species} has a too high BST for Shitmons.`, `(You are limited to Pok\u00e9mon with 600 BST or lower)`];
+			if (set.moves.length > 1 || this.getMove(set.moves[0]).id !== 'metronome') return [`${template.species} have illegal moves.`, `(You can only have Metronome in the moveset)`];
```

Comment:
While this works, set restrictions are normally handled similarly to this:
```javascript
if (set.moves) {
	if (set.moves.length > 1) problems.push(`${template.name} (${template.species}) can only have one moveslot.`);
	for (const moveid of set.moves) {
		let move = this.getMove(move);
		if (move.id !== 'metronome') problems.push(`${template.name} (${template.species})'s move ${move.name} is banned.`, `Only Metronome is allowed in Shitmons`)
	}
}
```

Reply:
I forgot to mention, but by doing this, if someone for some reason puts Metronome in the third moveslot instead of the first, it wont fail validation

Reply:
Looping through seems way overkill when all that's required is making sure there's only one move in the moveset, and that move is Metronome. It's on index 0 no matter which slot its put in into the teambuilder, and no other move is used.